### DISCUSSION TITLE
Outdated base image in kernel-py Dockerfile

### DISCUSSION THREAD

**Discussion by Chiragasourabh** [jupyter/enterprise_gateway#1276] [Q&A] [answered]
Any reason why kernel-py Dockerfile is still referencing scipy-notebook:2022-01-24 base image (why the version is hard coded in the first place?)

It has been more than a year since releasing the scipy-notebook:2022-01-24 image and the packages in it are outdated (to mention, the latest version of conda is 23.1.0 but scipy-notebook:2022-01-24 has conda 4.8.2)

**Comment by kevin-bates** [comment, marked as answer]
Hello @Chiragasourabh,

> Any reason why kernel-py Dockerfile is still referencing scipy-notebook:2022-01-24 base image (why the version is hard coded in the first place?)

We prefer pinned base images to ensure a certain degree of reproducibility (and stability).  Since we install updated kernels, we really don't leverage much from the base image.  

> It has been more than a year since releasing the scipy-notebook:2022-01-24 image and the packages in it are outdated (to mention, the latest version of conda is 23.1.0 but scipy-notebook:2022-01-24 has conda 4.8.2)

If you feel there are justified reasons to update the base images, please feel free to open a pull request.  I only ask that _all_ base images deriving from `docker-stacks` be updated to the same date-based tag.

**Comment by WalterMinorsJR2** [comment]
Is there any specific reason why the version is hard-coded in the first place? It would be great to update the base image to ensure that the packages are up to date and to avoid any potential security vulnerabilities.

**Comment by kevin-bates** [comment]
> Is there any specific reason why the version is hard-coded in the first place? It would be great to update the base image to ensure that the packages are up to date and to avoid any potential security vulnerabilities.

From above: We prefer pinned base images to ensure a certain degree of reproducibility (and stability). 

We should probably consider updating base images more frequently (perhaps on minor release boundaries?) but should not (IMO) use `latest` base images for (de)stability reasons.  Of course, should we become aware of a security issue in one of the referenced base images, then we should update that reference as soon as possible.
